### PR TITLE
Add Release Notes for Java Version 3.49

### PR DESCRIPTION
**Pull request by MatKuhr**
## What Has Changed?

Title

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/release-notes.mdx [hunk 1]
```diff
@@ -55,6 +55,48 @@ To continue using the latest features outlined in the release notes for version
...
 
+## 3.49.0 - July 15, 2021
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.49.0) | [Javadoc](https://help.sap.com/doc/55bb1642ea614bf582d514ed34401136/1.0/en-US/index.html)
+
+### Known issues
+
+- The module `cxf` uses the Apache CXF dependencies of version `3.3.10` which is affected by vulnerabilities [CVE-2021-30468](https://nvd.nist.gov/vuln/detail/CVE-2021-30468).
+  Currently, this module is not consumed in other Cloud SDK modules.
+  So, there is no harm during the runtime of any BTP application unless this module is directly consumed.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:66]: Why is this module there then? Should we just drop it? This abstract explains the issue but doesn't give a why.

**Review by artemkovalyov — COMMENTED** [review]
I didn't get the point about the vulnerability issue. Can we rephrase it somehow?

File: docs/java/release-notes.mdx (hunk 1 above)
Comment (MatKuhr) [reply docs/java/release-notes.mdx:66]: @KavithaSiva I think the sentence is missing which SDK module is affected? Could you maybe clarify the statement slightly?

Comment (KavithaSiva) [reply docs/java/release-notes.mdx:66]: The affected SDK module is `frameworks/cxf` module. This module is currently only consumed in our integration test module and not consumed in any other modules.

Comment (MatKuhr) [reply docs/java/release-notes.mdx]: I clarified it by adding the group ID

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -55,6 +55,48 @@ To continue using the latest features outlined in the release notes for version
...
 -->
 
+## 3.49.0 - July 15, 2021
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.49.0) | [Javadoc](https://help.sap.com/doc/55bb1642ea614bf582d514ed34401136/1.0/en-US/index.html)
+
+### Known issues
+
+- The module `com.sap.cloud.sdk.frameworks:cxf` uses the Apache CXF dependencies of version `3.3.10` which is affected by vulnerabilities [CVE-2021-30468](https://nvd.nist.gov/vuln/detail/CVE-2021-30468).
+  Currently, this module is not consumed in other Cloud SDK modules.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:65]:
```suggestion
  Currently, this module is not consumed in other Cloud SDK modules. It's used internally for integration tests.
```

File: docs/java/release-notes.mdx [hunk 3]
```diff
@@ -55,6 +55,48 @@ To continue using the latest features outlined in the release notes for version
...
+  So, there is no harm during the runtime of any BTP application unless this module is directly consumed.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:66]:
```suggestion
  Use of the SAP Cloud SDK doesn't compromise your application runtime unless you consume this module in your code directly.
```

**Review by artemkovalyov — APPROVED** [review]
Check my rephrasing. LGTM

**Pull request merged**